### PR TITLE
save secret to secret store after provisioning

### PR DESCRIPTION
**Pull request by maxtaco**
- in both KEX and paper
- tests that break on master
- tested by hand, seems to fix the issue

### CODE REVIEW

**Comment by maxtaco** [comment]
r? @patrickxb 

CC: @gabriel, this should fix the issue you saw. Thanks for pointing it out and your patience!

File: go/engine/login_test.go [hunk 1]
```diff
@@ -229,22 +229,33 @@ func TestProvisionDesktop(t *testing.T) {
...
+		teng := NewTrackEngine(arg, tcY.G)
+		if err := RunEngine(teng, ctx); err != nil {
+			t.Fatal(err)
+		}
 	}
+
+	testTrack("t_alice")
+
+	// Make sure that we can still track without a passphrase
+	// after a similated service restart.  In other words, that
```
Comment (patrickxb) [inline go/engine/login_test.go]: simulated

**Comment by gabriel** [comment]
Thanks!

File: go/engine/login_provision.go [hunk 2]
```diff
@@ -200,7 +214,11 @@ func (e *loginProvision) deviceWithType(ctx *Context, provisionerType keybase1.D
 	f := func(lctx libkb.LoginContext) error {
 		// run provisionee
 		ctx.LoginContext = lctx
-		return RunEngine(provisionee, ctx)
+		err := RunEngine(provisionee, ctx)
+		if err == nil {
```
Comment (gabriel) [inline go/engine/login_provision.go]: I like to follow the pattern:

```go
if err != nil {
  return err
}

return saveToSecretStore(e.G(), lctx, e.arg.User.GetNormalizedName(), provisionee.GetLKSec())

File: go/engine/login_provision.go [hunk 3]
```diff
@@ -248,6 +266,7 @@ func (e *loginProvision) paper(ctx *Context, device *libkb.Device) error {
 			// not a fatal error, session will stay in memory
 			e.G().Log.Warning("error saving session file: %s", err)
 		}
+		saveToSecretStore(e.G(), lctx, e.arg.User.GetNormalizedName(), e.lks)
```
Comment (gabriel) [inline go/engine/login_provision.go]:
```
return saveToSecretStore(e.G(), lctx, e.arg.User.GetNormalizedName(), e.lks)
```

**Comment by gabriel** [comment]
I think you are masking any errors from saveToSecretStore. Would be nice to fail if that fails, yes?

File: go/engine/login_provision.go (hunk 2 above)
Comment (gabriel) [inline go/engine/login_provision.go]: or even in this case:

```
if err := RunEngine(provisionee, ctx); err != nil {
  return err
}

return saveToSecretStore(e...)
```

**Comment by maxtaco** [comment]
it was intentional to not kill provisioning if we failed in this case.  for the other platforms, it's probably not the right thing to do, to fail the provisioning...  

we can reevaluate.  what happens if we return an error because the secret store step failed --- is the client half-provisioned?

**Comment by gabriel** [comment]
Yeah I think that's ok. I would probably error by default and then special case platforms that don't do secret store. I get nervous about silent failures but I suppose if it's non-fatal it makes sense.

I wonder down the road we can have non-fatal error classes that get reported like errors (to the user who may report it, and will fail tests maybe) but don't necessarily stop provisioning.

**Pull request merged**